doc/nrf/gs_programming.rst

```diff
@@ -119,3 +119,78 @@ After completing the :ref:`manual <build_environment_cli>` or :ref:`automatic <g
       The ``west flash`` command automatically resets the kit and starts the application.
 
 For more information on building and programming using the command line, see the Zephyr documentation on :ref:`zephyr:west-build-flash-debug`.
+
+.. _gs_programming_output:
+
+Output build files
+******************
+
+The building process produces each time an *image file*.
+
+.. output_build_files_info_start
+
+The image file, also referred to as *executable*, *program*, or *elf file*, consists of pieces of code and data that are identified by image-unique names recorded in a single symbol table.
+The symbol table exists as metadata in a :file:`.elf` or :file:`.exe` file and is not included when the image is converted to a hexidecimal file for programming.
+Instead, a linker places the code and data at their addresses.
+
+Only images require this linking process.
```

Comment:
I'm not sure what these two lines try to convey. Can anyone help? Unless I misunderstand something this seems to be pretty technical information that is either known for users or it cannot be explained in two sentences, so probably can be removed.

Reply:
I wasn't sure if we need this when I was moving these lines from below. Looks like an introduction to why we can have multi-image builds.

Reply:
Looks like the original author (Torsten) is absent, so we can leave it as it is, but reading this for the first time I find it rather confusing and not crucial to grasp the idea of multi-image builds.

Reply:
@Damian-Nordic i'm not the author of this. I'm the the committer (because I merged the PR), the authors are:
@kyberdin and @FrancescoSer 
Ref: https://github.com/nrfconnect/sdk-nrf/commit/2f76e95ad435161135111e16b00f2d0eb04338b3

Reply:
Agreed with @Damian-Nordic, this is unnecessary information for the user and probably just leads to more confusion. The original docs were worded poorly to begin with and this doesn't convey the actual informative intent, but users that care about that level of detail probably already know it anyway.

This line and Line 137 are probably just better off removed to keep the info here simpler and more cohesive.